sakai-evaluation-tool/src/java/org/sakaiproject/evaluation/tool/reporting/XLSReportExporter.java

```diff
@@ -79,223 +95,627 @@ public void setMessageLocator(MessageLocator locator) {
+        List<List<String>> courseRelatedResponses = new ArrayList<List<String>>();
+        Map<Long, Map<User, List<EvalAnswer>>> answerMap = new HashMap<Long, Map<User, List<EvalAnswer>>>();
+        for( Long responseID : responseIDs )
+        {
+            // Dump the (course related) data into a list of strings representing a spreadsheet row (so it can be sorted)
+            List<String> row = new ArrayList<String>();
+            row.add( SECTION_OR_SITE_COLUMN_NUM, responseAggregator.getCommaSeparatedGroupNames(
+                    new String[] { tidl.getAnswersByResponseId( responseID ).get( 0 ).getResponse().getEvalGroupId() } ) );
+            row.add( RESPONSE_ID_COLUMN_NUM, responseID.toString() );
+
+            // Add the response ID to the answer map
+            answerMap.put( responseID, new HashMap<User, List<EvalAnswer>>() );
+
+            // Loop through the data template items...
+            int questionCounter = 0;
+            for( DataTemplateItem dti : dtiList )
+            {
+                if( !instructorViewAllResults                                                       // If the eval is so configured,
+                      && !commonLogic.isUserAdmin( currentUserId )                                  // and currentUser is not an admin
+                      && !currentUserId.equals( evalOwner )                                         // and currentUser is not the eval creator
+                      && !EvalConstants.ITEM_CATEGORY_COURSE.equals( dti.associateType )            // and the associate type is not 'course'
+                      && !currentUserId.equals( commonLogic.getEvalUserById( dti.associateId ).userId ) )
+                {
+                    continue;                                                                       // skip items that aren't for the current user
+                }
+
+                // If it's an instructor related item... 
+                EvalAnswer answer = dti.getAnswer( responseID );
+                if( EvalConstants.ITEM_CATEGORY_ASSISTANT.equals( dti.associateType ) || EvalConstants.ITEM_CATEGORY_INSTRUCTOR.equals( dti.associateType ) )
+                {
+                    // If the answer is NOT null (it would be null for an instructor from a different section than the evaluator)
+                    if( answer != null )
+                    {
+                        // Get the instructor
+                        User instructor = null;
+                        try { instructor = userDirectoryService.getUser( answer.getAssociatedId() ); }
+                        catch( UserNotDefinedException ex ) { continue; }
+
+                        // If the answer map has a list of answers for this response and this instructor, add the answer to the list
+                        if( answerMap.get( responseID ).containsKey( instructor ) )
+                        {
+                            answerMap.get( responseID ).get( instructor ).add( answer );
```

Comment:
This will work:
Map<User, List<EvalAnswer>> responseAnswers = answerMap.get(responseID);
List<EvalAnswer> instructorAnswers = responseAnswers.get(instructor);
if (instructorAnswers == null)
{
    instructorAnswers = new ArrayList<>();
    answerMap.put(instructorAnswers);
}
instructorAnswers.add(answer);